src/irr/core/CFixedCapacityDoublyLinkedList.h

```diff
@@ -38,73 +38,84 @@ namespace irr {
 		template<typename Value>
 		class DoublyLinkedList
 		{
-			typedef Snode<Value>* iteratorptr_t;
-
+			typedef uint32_t node_address;
+			
 		private:
 			void* reservedSpace;
 			PoolAddressAllocator<uint32_t> alloc;
-			iteratorptr_t p_begin;
-			iteratorptr_t p_end;
+			node_address m_back;
+			node_address m_begin;
+			Snode<Value>* m_array;
 			uint32_t cap;
 
+			#define invalid_iterator alloc.invalid_address	//0xdeadbeef
+
+		public:
+
 			inline void popBack()
 			{
-				if(p_end->prev != nullptr)
-					p_end->prev->next = nullptr;
-				iteratorptr_t temp = p_end;
-				p_end = p_end->prev;
-				alloc.free_addr(reinterpret_cast<uint32_t>(temp), 1u);
+				if(m_back->prev != invalid_iterator)
+					m_back->prev->next = invalid_iterator;
+				uint32_t temp = m_end;
+				m_end = m_back->prev;
```

Comment:
what is `m_end` ?

Reply:
also you cant just use `m_end->` and `m_begin->` , you'd need to implement a tiny allocator wrapper struct over the `uint32_t` to be able to do that.... or just use `m_array[m_end].` instead.